pep-9999.rst

```diff
@@ -47,13 +47,57 @@ required to interpret all standard constructs, and additionally a type checker
 can support extra constructs not described in this PEP. All deviations from
 this PEP should be documented.
 
+Syntax and Directory Structure
+==============================
+
+Type stubs are syntactically valid Python files with a ``.pyi`` suffix.
+Type checkers are expected to support the syntax of the latest Python
+version. Type stubs should be parseable with the latest released Python
```

Comment:
Should the sentence about type checkers say something about the timeframe of expected support for the latest version (within six months of release) for symmetry with the type stub requirement?

Reply:
I think it makes sense that type checkers aim to support a new Python version as soon as it comes out, so people type checking their code can use a new Python version when it becomes available. But I wonder whether it makes sense to reduce the six month to something like three?

Reply:
What strikes me as odd about the current recommendation is that saying both that type checkers should immediately support the latest version and that stubs should give checkers a six month grace period means that there's no incentive for the first timeline to be followed.

I think either allowing stubs to immediately use new syntax or explicitly giving type checkers a grace period (whether that's six months or something shorter) would make sense to me.

Reply:
I think I have described it backwards in the PR: Stubs can start to use new syntax immediately, but can use obsolete syntax up until 6 months after the release of a Python version. For the async example: Type checkers should have supported the `async` keyword in Python 3.7 code with the release of 3.7. Stubs can use `async` as a keyword immediately as well, but can still use `async` as an attribute up until 6 months after the release of 3.7. (Of course in the meantime the stubs can't be used to check 3.7 code.) Does this make more sense?

Reply:
Ahhh, so the intent is that stubs have up to six months to become compatible with the newest version, rather than having to stay compatible with the previous version for that time period? In that case, yeah, I think the PR has described things the wrong way around.